src/Mono.Android/Android.Runtime/JNIEnvInit.cs

```diff
@@ -79,7 +79,13 @@ static Type TypeGetType (string typeName) =>
 			JniType.GetCachedJniType (ref jniType, className);
 
 			ReadOnlySpan<char> methods = new ReadOnlySpan<char> ((void*) methods_ptr, methods_len);
-			((AndroidTypeManager)androidRuntime!.TypeManager).RegisterNativeMembers (jniType, type, methods);
+			if (androidRuntime!.TypeManager is AndroidTypeManager typeManager) {
```

Comment:
This runtime check shouldn't be needed.

There is already a `JniRuntime.JniTypeManager.RegisterNativeMembers()` method which takes `ReadOnlySpan<char>`: https://github.com/dotnet/java-interop/blob/7b45b16627f6305c5f02cda0097d528cbcf2893e/src/Java.Interop/Java.Interop/JniRuntime.JniTypeManager.cs#L461-L468

We just don't `override` it at the moment:

https://github.com/dotnet/android/blob/d81cde9c4619312898dd0415a9ba43e6a77648ac/src/Mono.Android/Android.Runtime/AndroidRuntime.cs#L499

Perhaps all we need to do is add `override`?

Reply:
That seems better, and still builds locally.